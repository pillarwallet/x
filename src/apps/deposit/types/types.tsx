export type BalanceInfo = {
  type?: 'BalanceInfo';
  name: string;
  chain: string;
  address: string;
  decimals: number;
  symbol: string;
  logoURI: string;
  balance: string;
};

export type AddedAssets = {
  type?: 'AddedAsset';
  assetType: 'token' | 'nft';
  tokenId?: string;
  chain: string;
  tokenAddress: string;
  balance: number;
};

export type Network =
  | 'ethereum'
  | 'polygon'
  | 'gnosis'
  | 'base'
  | 'bnb smart chain'
  | 'optimism'
  | 'arbitrum';

export type TokenList = {
  address: string;
  chainId: number;
  decimals: string;
  name: string;
  symbol: string;
  logoURI: string;
};

export type BalancesByChain = {
  [chain: string]: BalanceInfo[];
};
