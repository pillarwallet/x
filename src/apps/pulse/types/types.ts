export type TransactionStatusState =
  | 'Starting Transaction'
  | 'Transaction Pending'
  | 'Transaction Complete'
  | 'Transaction Failed';

export type TransactionStep =
  | 'Submitted'
  | 'Pending'
  | 'ResourceLock'
  | 'Completed';

export type StepStatus = 'completed' | 'pending' | 'failed' | 'inactive';

export interface TokenInfo {
  symbol: string;
  name: string;
  logo: string;
  address?: string;
}

export interface SellOffer {
  tokenAmountToReceive: number;
  minimumReceive: number;
}

export interface PayingToken {
  totalUsd: number;
  name: string;
  symbol: string;
  logo: string;
  actualBal: string;
  totalRaw: string;
  chainId: number;
  address: string;
}

export interface TokenDetails {
  symbol: string;
  name: string;
  address: string;
  chainId: number;
  amount: string;
  logo: string;
  type: 'BUY_TOKEN' | 'SELL_TOKEN';
}

export interface BuyModeDetails {
  usdAmount: string;
  payingTokens: PayingToken[];
  totalPayingUsd: number;
}

export interface TransactionStatusConfig {
  icon: string;
  containerClasses: string;
  iconClasses: string;
  color: string;
}

export interface ButtonConfig {
  bgColor: string;
  textColor: string;
  borderColor: string;
  label: string;
}

export interface ProgressStepConfig {
  label: string;
  order: number;
}

export interface TransactionStatusProps {
  closeTransactionStatus: () => void;
  userOpHash: string;
  chainId: number;
  gasFee?: string;
  // Transaction data from PreviewSell/PreviewBuy
  isBuy?: boolean;
  sellToken?: TokenInfo | null;
  buyToken?: TokenInfo | null;
  tokenAmount?: string;
  sellOffer?: SellOffer | null;
  payingTokens?: PayingToken[];
  usdAmount?: string;
  // Externalized polling state
  currentStatus: TransactionStatusState;
  errorDetails: string;
  submittedAt?: Date;
  pendingCompletedAt?: Date;
  blockchainTxHash?: string;
  resourceLockTxHash?: string;
  completedTxHash?: string;
  completedChainId?: number;
  resourceLockChainId?: number;
  resourceLockCompletedAt?: Date;
  isResourceLockFailed?: boolean;
}

export interface TransactionDetailsProps {
  onDone: () => void;
  userOpHash: string;
  chainId: number;
  status: TransactionStatusState;
  // Transaction data from PreviewSell/PreviewBuy
  isBuy?: boolean;
  sellToken?: TokenInfo | null;
  buyToken?: TokenInfo | null;
  tokenAmount?: string;
  sellOffer?: SellOffer | null;
  payingTokens?: PayingToken[];
  usdAmount?: string;
  submittedAt?: Date;
  pendingCompletedAt?: Date;
  resourceLockCompletedAt?: Date;
  txHash?: string;
  gasFee?: string;
  errorDetails?: string;
  resourceLockTxHash?: string;
  completedTxHash?: string;
  resourceLockChainId?: number;
  completedChainId?: number;
  isResourceLockFailed?: boolean;
}

export interface TransactionInfoProps {
  status: TransactionStatusState;
  userOpHash: string;
  txHash?: string;
  chainId: number;
  gasFee?: string;
  completedAt?: Date;
  // Buy-specific fields
  isBuy?: boolean;
  resourceLockTxHash?: string;
  resourceLockChainId?: number;
  completedTxHash?: string;
  completedChainId?: number;
}

export interface ProgressStepProps {
  step: TransactionStep;
  status: StepStatus;
  label: string;
  isLast?: boolean;
  showLine?: boolean;
  lineStatus?: StepStatus;
  timestamp?: string | React.ReactNode;
}

export interface TransactionErrorBoxProps {
  technicalDetails?: string;
}

export interface UseClickOutsideOptions {
  ref: React.RefObject<HTMLElement>;
  callback: () => void;
  condition: boolean;
}

export interface UseKeyboardNavigationOptions {
  onEscape: () => void;
  onEnter?: () => void;
  enabled?: boolean;
}

export interface TechnicalDetails {
  transactionType: 'BUY' | 'SELL';
  transactionHash: string;
  hashType: 'bidHash' | 'userOpHash';
  chainId: number;
  status: TransactionStatusState;
  timestamp: string;
  accountAddress: string;
  token: TokenDetails | null;
  sellOffer: SellOffer | null;
  buyMode: BuyModeDetails | null;
  transactionHashes: {
    [key: string]: string;
  };
  chains: {
    mainChainId: number;
    resourceLockChainId: string | number;
    completedChainId: string | number;
  };
  timestamps: {
    [key: string]: string;
  };
  error: {
    details: string;
    isResourceLockFailed: boolean;
    failureStep: string;
  };
  gas: {
    fee: string;
  };
  stepStatus: {
    [key: string]: StepStatus;
  };
}
