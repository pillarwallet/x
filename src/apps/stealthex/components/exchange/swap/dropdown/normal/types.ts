
import type React from 'react';
import { StructuredCurrency } from '../../../../../type';
import { CurrencyData } from '../../../../../lib/backend/api';

export type ClickCallback = (currency: CurrencyData) => void;

export type AvailableCurrency =
  | {
    availableCurrency?: Omit<StructuredCurrency, 'fiat'> & {
      all?: CurrencyData[];
    };
    type: 'get';
  }
  | {
    availableCurrency?: StructuredCurrency & {
      all?: CurrencyData[];
    };
    type: 'send';
  };

export type NormalDropdownProps = AvailableCurrency & {
  loading?: boolean;
  selectedCurrency: CurrencyData | null;
  searchContext: string;
  fiat?: boolean;
  onClick?: ClickCallback;
  components?: {
    NotFound?: React.FC<React.PropsWithChildren>;
    Group?: React.FC<React.PropsWithChildren>;
    Item?: React.FC<
      React.PropsWithChildren<
        | { loading: true }
        | {
          loading?: false;
          currency: CurrencyData;
          selected?: boolean;
          onClick?: () => void;
        }
      >
    >;
  };
  groupHeight?: number;
  itemHeight?: number;
  maxHeight?: number;
  embed?: boolean;
};
