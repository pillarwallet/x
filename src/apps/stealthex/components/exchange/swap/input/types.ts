
import { CurrencyData } from '../../../../lib/backend/api';
import type { AvailableCurrency } from '../dropdown/normal/types';

export type SavedAvailableCurrency =
  | {
      floatingDropdown: true;
      availableCurrency: CurrencyData[];
      type: 'get' | 'send';
    }
  | ({
      floatingDropdown?: false;
    } & AvailableCurrency);

export type InputProps = {
  currency: CurrencyData;
  value?: string;
  loading?: boolean;
  loadingCurrency?: boolean;
  error?: string | JSX.Element;
  fiat?: boolean;
  disabled?: boolean;
  withLock?: boolean | null;
  widget?: boolean;
  fixed?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  onFixedChange?: (fixed: boolean) => void;
  onCurrencySelect?: (currency: CurrencyData) => void;
  pressable?: boolean;
} & SavedAvailableCurrency;
