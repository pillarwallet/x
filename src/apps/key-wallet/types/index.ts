export interface Asset {
  id: number;
  name: string;
  symbol: string;
  logo: string;
  balance: number;
  decimals: number;
  price: number;
  price_change_24h: number;
  contract: string;
  chainId: number;
  chainName: string;
  usdBalance: number;
}

export interface SendFormData {
  amount: string;
  recipient: string;
}

export interface TransactionStatus {
  hash: string;
  chainId: number;
  status: 'pending' | 'success' | 'failed';
  timestamp: number;
}

