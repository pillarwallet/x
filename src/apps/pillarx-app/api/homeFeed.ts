/* eslint-disable no-console */
import { createApi, fetchBaseQuery, retry } from '@reduxjs/toolkit/query/react';

// store
import { addMiddleware } from '../../../store';

// types
import { ApiResponse, WalletData } from '../../../types/api';

// utils
import { CompatibleChains } from '../../../utils/blockchain';

const chainIds =
  process.env.REACT_APP_USE_TESTNETS === 'true'
    ? [11155111]
    : CompatibleChains.map((chain) => chain.chainId);

const chainIdsQuery = chainIds.map((id) => `chainIds=${id}`).join('&');

export const homeFeedApi = createApi({
  reducerPath: 'homeFeedApi',
  baseQuery: fetchBaseQuery({
    baseUrl:
      process.env.REACT_APP_USE_TESTNETS === 'true'
        ? 'https://feed-nubpgwxpiq-uc.a.run.app'
        : 'https://feed-7eu4izffpa-uc.a.run.app',
  }),
  endpoints: (builder) => ({
    getTilesInfo: builder.query<ApiResponse, { page: number; address: string }>(
      {
        query: ({ page, address }) =>
          `?page=${page}&address=${address}&${chainIdsQuery}&testnets=${process.env.REACT_APP_USE_TESTNETS || 'true'}`,
      }
    ),
  }),
});

// maxRetries: 5 is the default, and can be omitted. Shown for documentation purposes.
const staggeredBaseQuery = retry(
  fetchBaseQuery({
    baseUrl:
      process.env.REACT_APP_USE_TESTNETS === 'true'
        ? 'https://walletportfolio-nubpgwxpiq-uc.a.run.app'
        : 'https://walletportfolio-7eu4izffpa-uc.a.run.app',
  }),
  {
    maxRetries: 5,
  }
);

export const walletPortfolioTileApi = createApi({
  reducerPath: 'walletPortfolioTileApi',
  baseQuery: staggeredBaseQuery,
  endpoints: (builder) => ({
    getWalletInfo: builder.query<WalletData, { address: string }>({
      query: ({ address }) =>
        `?address=${address}&${chainIdsQuery}&testnets=${process.env.REACT_APP_USE_TESTNETS || 'true'}`,
    }),
  }),
});

/**
 * Add this to the store
 */
addMiddleware(homeFeedApi);
addMiddleware(walletPortfolioTileApi);

export const { useGetTilesInfoQuery } = homeFeedApi;
export const { useGetWalletInfoQuery } = walletPortfolioTileApi;
