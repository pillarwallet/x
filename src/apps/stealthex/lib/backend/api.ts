import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
import { RequestResponse } from './type';
import { ExchangeInfo } from '../../type';
import { ProviderName } from '../providers';

export const baseUrl = 'https://stealthex.io';

export type Abortable = { signal?: AbortSignal }

export type IsFixed = {
    isFixed: boolean;
};

export type CreateExchangeParams = {
    currencyFrom: string;
    currencyTo: string;
    address: string;
    extraId?: string;
    amountTo?: string;
    amountFrom?: string;
    referral?: string;
    fixed?: boolean;
    timezoneOffset: number;
    isWidget?: boolean;
    partnerId?: string;
    rate_id?: string;
    refundAddress?: string;
    provider?: ProviderName;
};
export type Estimate = {
    estimate: string | null;
    rate_id?: string;
};

export type Range = {
    min_amount?: string;
    max_amount?: string;
    message?: string;
};

export type EstimateProps = {
    from: string | null;
    to: string | null;
    amount?: number;
    amount_to?: number;
    fixed: boolean;
    provider?: 'simplex' | 'mercuryo';
    signal?: AbortSignal;
};

export type RangeProps = {
    to: string | null;
    from: string | null;
    fixed?: boolean;
    fiat?: boolean;
    reverse?: boolean;
};

export type CoinImage = {
    id?: string;
    mimetype?: string;
    size?: number;
    filename: string;
    originalFilename?: string;
    encoding?: string;
    _meta?: boolean;
};

export type Post = {
    id: number;
    date: string;
    link: string;
    title: {
        rendered: string;
    };
    excerpt: {
        rendered: string;
    };
    featured_media: number;
    categories: string[];
    tags: number[];
    modified?: string;
};

export type FormattedPost = Post & {
    image: string;
};

export type CoinData = {
    change24h?: string;
    changepct24h?: string;
    high24h?: string;
    low24h?: string;
    mktcap?: string;
    price: string;
    network: string | null;
    supply?: string;
    volume24h?: string;
    _id: string;
    image: string;
    symbol: string;
    slug: string;
    name: string;
    about?: string;
    order?: number;
    colorIcon?: CoinImage;
    seoImage?: CoinImage;
    disableTradingView: false;
    priceChart: {
        type: 'third_party';
        symbol: string;
    };
    relatedArticles: FormattedPost[];
};

export type CurrencyData = {
    symbol: string;
    slug?: string;
    has_extra_id: boolean;
    extra_id: string | null;
    is_fiat?: boolean;
    name: string;
    network: string | null;
    warnings_from: string[];
    warnings_to: string[];
    validation_address: string | null;
    validation_extra: string | null;
    address_explorer: string | null;
    tx_explorer: string | null;
    image: string;
};

export type PairsTo = {
    popular: CurrencyData[];
    other: CurrencyData[];
};

export type PairsFrom = {
    popular: CurrencyData[];
    other: CurrencyData[];
    fiat: CurrencyData[];
};

export const exchange = createApi({
    reducerPath: 'exchangeInfo',
    baseQuery: fetchBaseQuery({ baseUrl }),
    endpoints: (builder) => ({
        exchangeInfo: builder.query<{ data: ExchangeInfo, signature: string }, string>({
            query: (exchangeId) => ({ url: `/api/exchange/v2/${exchangeId}` })
        }),
        createExchange: builder.mutation<{ data: ExchangeInfo, signature: string }, CreateExchangeParams>({
            query: ({ ...params }) => {
                const { currencyFrom, currencyTo, address, extraId, amountFrom, amountTo, isWidget, partnerId, refundAddress, ...otherParams } = params;
                return ({
                    url: '/api/exchange/v2/create', method: 'POST', body: {
                        currency_from: currencyFrom,
                        currency_to: currencyTo,
                        address_to: address,
                        extra_id_to: extraId,
                        amount_from: amountFrom,
                        amount_to: amountTo,
                        partner_id: partnerId,
                        is_widget: isWidget,
                        refund_address: refundAddress,
                        refund_extra_id: '',
                        ...otherParams,
                    }
                })
            },
        }),
        estimate: builder.query<Estimate, EstimateProps>({
            query: ({ amount, amount_to, from, to, fixed, signal }) => ({ url: `/api/exchange/estimate/v3/${from}/${to}`, params: { amount, fixed, amount_to }, signal })
        }),
        fiatEstimate: builder.query<Estimate, EstimateProps>({
            query: ({ amount, from, to, signal, provider }) => ({ url: `/api/exchange/estimate/v3/${from}/${to}`, params: { amount, fixed: false, provider }, signal })
        }),
        fiatCoins: builder.query<RequestResponse<false, CoinData[]>, Abortable>({
            query: ({ signal }) => ({ url: '/api/coins/fiat', signal, params: { timeout: false } })
        }),
        searchCurrency: builder.query<RequestResponse<false, CurrencyData>, { query: string, signal?: AbortSignal, onlySymbolSearch: boolean, }>({
            query: ({ signal, query, onlySymbolSearch }) => ({ url: '/api/exchange/searchCurrency', signal, params: { query, only_symbol_search: onlySymbolSearch } })
        }),
        range: builder.query<Range, RangeProps & { signal?: AbortSignal, enabled?: boolean }>({
            query: ({ to, from, signal, ...searchParams }) => ({ url: `/api/exchange/range/${from}/${to}`, params: searchParams, signal })
        }),
        pairsFrom: builder.query<PairsFrom, { coin: string | null } & Abortable>({
            query: ({ coin, signal }) => ({ url: `/api/exchange/pairsfrom/${coin}`, signal })
        }),
        pairsTo: builder.query<PairsTo, { coin: string | null } & Abortable>({
            query: ({ coin, signal }) => ({ url: `/api/exchange/pairs/${coin}`, signal })
        }),
        isFixed: builder.query<IsFixed, { from: string, to: string } & Abortable>({
            query: ({ from, to, signal }) => ({ url: `/api/exchange/is-fixed/${from}/${to}`, signal })
        })
    })
})

// export const estimate = createApi({
//     reducerPath: 'estimate',
//     baseQuery: fetchBaseQuery({baseUrl}),
//     endpoints: (builder) => ({
//         estimateInfo: builder.mutation<RequestResponse<true,Estimate>,CreateExchangeParams & {signal: AbortSignal}>({
//             query: ({currencyTo, currencyFrom,signal,  ...searchParams }) => ({url: `/api/exchange/estimate/v3/${currencyFrom}/${currencyTo}/?amount=${searchParams.}`,signal, method: 'POST'  })
//         })
//     })
// })


export const { useExchangeInfoQuery, useCreateExchangeMutation, useFiatEstimateQuery, useFiatCoinsQuery, useSearchCurrencyQuery, useRangeQuery, useEstimateQuery, usePairsFromQuery, usePairsToQuery, useIsFixedQuery } = exchange;