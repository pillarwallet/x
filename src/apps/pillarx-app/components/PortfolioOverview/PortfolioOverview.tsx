import { useWalletAddress } from '@etherspot/transaction-kit';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';

// utils
import { getAllUniqueBlockchains } from '../../utils/blockchain';

// types
import { WalletData, WalletPortfolioData } from '../../../../types/api';

// images
import BlendIcon from '../../images/blend-icon.svg';
import RoutingIcon from '../../images/routing-icon.svg';

// components
import SkeletonLoader from '../../../../components/SkeletonLoader';
import Tags from '../Tags/Tags';
import TileContainer from '../TileContainer/TileContainer';
import TokensHorizontalList from '../TokensHorizontalList/TokensHorizontalList';
import TokensPercentage from '../TokensPercentage/TokensPercentage';
import Body from '../Typography/Body';
import H1 from '../Typography/H1';
import WalletAddressOverview from '../WalletAdddressOverview/WalletAddressOverview';
import WalletConnectModal from '../WalletConnectModal/WalletConnectModal';

type PortfolioOverviewProps = {
  data: WalletData | undefined;
  isDataLoading: boolean;
};

const PortfolioOverview = ({ data, isDataLoading }: PortfolioOverviewProps) => {
  const [t] = useTranslation();
  const [isWalletConnectModalOpen, setIsWalletConnectModalOpen] =
    useState<boolean>(false);
  const accountAddress = useWalletAddress();
  const { data: dataPortlioOverview } = data || {};
  const dataWallet = dataPortlioOverview as WalletPortfolioData | undefined;
  const { percentage_change: percentageChange = 0 } =
    dataWallet?.total_pnl_history?.['24h'] || {};

  const numberOfTokens = dataWallet?.assets?.length || 0;

  const allBlockchains =
    dataWallet?.assets?.map((asset) => asset.asset.blockchains).flat() || [];

  const allBlockchainsLogos =
    dataWallet?.assets
      ?.map((asset) => (asset.asset.logo ? asset.asset.logo : 'random-avatar'))
      .flat() || [];

  const numberOfBlockchains =
    getAllUniqueBlockchains(allBlockchains).length ?? 0;

  if (!data || isDataLoading) {
    return (
      <TileContainer
        id="wallet-portfolio-tile-loader"
        className="p-10 gap-20 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23] mobile:flex-col mobile:gap-4"
      >
        <div className="flex flex-col justify-between">
          <Body className="animate-pulse mobile:my-4">
            Wallet portfolio data loading...
          </Body>
          <div className="mobile:border mobile:border-medium_grey mobile:rounded-[10px] mobile:p-4 mobile:w-full">
            <SkeletonLoader $height="77px" $width="180px" $radius="10px" />
          </div>
        </div>
        <div className="flex w-full flex-col items-end justify-end mobile:justify-normal gap-5">
          <div className="flex w-full gap-4 justify-end mobile:flex-row tablet:flex-col tablet:items-end mobile:flex-row mobile:justify-between mobile:items-start">
            <SkeletonLoader $height="36px" $width="150px" $radius="10px" />
            <SkeletonLoader $height="36px" $width="190px" $radius="10px" />
          </div>
          <SkeletonLoader $height="36px" $width="80px" $radius="10px" />
        </div>
      </TileContainer>
    );
  }

  return (
    <>
      <TileContainer
        id="wallet-portfolio-tile"
        className="p-10 gap-20 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23] mobile:flex-col mobile:gap-4"
      >
        <div className="flex flex-col justify-between">
          <WalletAddressOverview address={accountAddress ?? ''} />
          <button
            type="button"
            className="max-w-fit p-2 rounded-md bg-purple_medium text-sm mb-2"
            onClick={() =>
              setIsWalletConnectModalOpen((prevState) => !prevState)
            }
          >
            Connect with WalletConnect
          </button>
          <div className="mobile:border mobile:border-medium_grey mobile:rounded-[10px] mobile:p-4 mobile:w-full">
            <Body className="text-purple_light mb-2">{t`title.totalBalance`}</Body>
            <div className="flex gap-4 items-end">
              <H1 className="text-[50px]">
                ${dataWallet?.total_wallet_balance?.toFixed(2) || 0}
              </H1>
              <TokensPercentage percentage={percentageChange} />
            </div>
          </div>
        </div>
        <div className="flex w-full flex-col items-end justify-end mobile:justify-normal gap-5">
          <div className="flex w-full gap-4 justify-end mobile:flex-row tablet:flex-col tablet:items-end mobile:flex-row mobile:justify-between">
            {numberOfTokens > 0 && (
              <Tags
                icon={BlendIcon}
                tagText={`${numberOfTokens} ${t`label.tokens`}`}
              />
            )}
            {numberOfBlockchains > 0 && (
              <Tags
                icon={RoutingIcon}
                tagText={`${t`helper.across`} ${numberOfBlockchains} ${numberOfBlockchains > 1 ? t`helper.chainSeveral` : t`helper.chainOne`}`}
              />
            )}
          </div>
          {allBlockchainsLogos.length > 0 && (
            <TokensHorizontalList logos={allBlockchainsLogos} />
          )}
        </div>
      </TileContainer>
      <WalletConnectModal
        isOpen={isWalletConnectModalOpen}
        onClose={() => setIsWalletConnectModalOpen((prevState) => !prevState)}
      />
    </>
  );
};

export default PortfolioOverview;
