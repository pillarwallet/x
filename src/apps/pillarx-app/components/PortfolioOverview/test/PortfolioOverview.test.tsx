import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

// types
import { WalletData } from '../../../../../types/api';

// components
import PortfolioOverview from '../PortfolioOverview';

const mockData: WalletData = {
  data: {
    wallet: 'wallet_address',
    wallets: [''],
    total_wallet_balance: 1000,
    assets: [],
    total_realized_pnl: 53,
    total_unrealized_pnl: 12,
    total_pnl_history: {
      '24h': { realized: 100, unrealized: 200, percentage_change: 1.0 },
    },
  },
};

const mockNoData: WalletData = {
  data: {
    wallet: undefined,
    wallets: [''],
    total_wallet_balance: undefined,
    assets: [],
    total_realized_pnl: undefined,
    total_unrealized_pnl: undefined,
    total_pnl_history: undefined,
  },
};

const mockLoading = false;

describe('<PortfolioOverview />', () => {
  it('renders correctly and matches snapshot', () => {
    const tree = renderer
      .create(<PortfolioOverview data={mockData} isDataLoading={mockLoading} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('displays loading skeleton when data is loading', () => {
    const tree = renderer
      .create(<PortfolioOverview data={undefined} isDataLoading />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders data correctly when not loading', () => {
    render(<PortfolioOverview data={mockData} isDataLoading={false} />);

    expect(screen.getByText('$1000.00')).toBeInTheDocument();
    expect(screen.getByText('1.00%')).toBeInTheDocument();
    expect(screen.queryByText('tokens')).not.toBeInTheDocument();
    expect(screen.queryByText('chains')).not.toBeInTheDocument();
    expect(screen.queryByTestId('tokens-list')).not.toBeInTheDocument();
  });

  it('calculates percentage change correctly', () => {
    render(<PortfolioOverview data={mockData} isDataLoading={false} />);

    expect(
      screen.getByText(
        `${mockData.data.total_pnl_history?.['24h']?.percentage_change.toFixed(2)}%`
      )
    ).toBeInTheDocument();
  });

  it('handles no data state correctly', () => {
    render(<PortfolioOverview data={mockNoData} isDataLoading={false} />);

    expect(screen.getByText('$0')).toBeInTheDocument();
    expect(screen.queryByText('tokens')).not.toBeInTheDocument();
    expect(screen.queryByText('chains')).not.toBeInTheDocument();
  });
});
