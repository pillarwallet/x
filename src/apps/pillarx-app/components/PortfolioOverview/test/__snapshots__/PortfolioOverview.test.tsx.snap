// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfolioOverview /> displays loading skeleton when data is loading 1`] = `
.c0 {
  height: 77px;
  width: 180px;
  animation: hLhBjP 1s linear infinite alternate;
  border-radius: 10px;
  margin-bottom: 0;
}

.c1 {
  height: 36px;
  width: 150px;
  animation: hLhBjP 1s linear infinite alternate;
  border-radius: 10px;
  margin-bottom: 0;
}

.c2 {
  height: 36px;
  width: 190px;
  animation: hLhBjP 1s linear infinite alternate;
  border-radius: 10px;
  margin-bottom: 0;
}

.c3 {
  height: 36px;
  width: 80px;
  animation: hLhBjP 1s linear infinite alternate;
  border-radius: 10px;
  margin-bottom: 0;
}

<div
  className="flex bg-container_grey rounded-2xl p-10 gap-20 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23] mobile:flex-col mobile:gap-4"
  id="wallet-portfolio-tile-loader"
>
  <div
    className="flex flex-col justify-between"
  >
    <p
      className="text-base font-medium animate-pulse mobile:my-4"
    >
      Wallet portfolio data loading...
    </p>
    <div
      className="mobile:border mobile:border-medium_grey mobile:rounded-[10px] mobile:p-4 mobile:w-full"
    >
      <div
        className="c0"
      />
    </div>
  </div>
  <div
    className="flex w-full flex-col items-end justify-end mobile:justify-normal gap-5"
  >
    <div
      className="flex w-full gap-4 justify-end mobile:flex-row tablet:flex-col tablet:items-end mobile:flex-row mobile:justify-between mobile:items-start"
    >
      <div
        className="c1"
      />
      <div
        className="c2"
      />
    </div>
    <div
      className="c3"
    />
  </div>
</div>
`;

exports[`<PortfolioOverview /> renders correctly and matches snapshot 1`] = `
<div
  className="flex bg-container_grey rounded-2xl p-10 gap-20 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23] mobile:flex-col mobile:gap-4"
  id="wallet-portfolio-tile"
>
  <div
    className="flex flex-col justify-between"
  >
    <div
      className="desktop:flex tablet:flex mobile:hidden mb-[54px] gap-2.5 items-center"
    >
      <div
        className="bg-medium_grey rounded-full w-10 h-10 overflow-hidden"
      >
        <svg
          className="rounded-md"
          data-testid="random-avatar"
          fill="none"
          role="img"
          viewBox="0 0 80 80"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height={80}
            id=":r0:"
            maskUnits="userSpaceOnUse"
            width={80}
            x={0}
            y={0}
          >
            <rect
              fill="#FFFFFF"
              height={80}
              width={80}
            />
          </mask>
          <g
            mask="url(#:r0:)"
          >
            <rect
              fill="#C271B4"
              height={80}
              width={80}
            />
            <path
              d="M32.414 59.35L50.376 70.5H72.5v-71H33.728L26.5 13.381l19.057 27.08L32.414 59.35z"
              fill="#C20D90"
              filter="url(#filter_:r0:)"
              transform="translate(4 4) rotate(276 40 40) scale(1.4)"
            />
            <path
              d="M22.216 24L0 46.75l14.108 38.129L78 86l-3.081-59.276-22.378 4.005 12.972 20.186-23.35 27.395L22.215 24z"
              fill="#92A1C6"
              filter="url(#filter_:r0:)"
              style={
                {
                  "mixBlendMode": "overlay",
                }
              }
              transform="translate(6 -6) rotate(54 40 40) scale(1.4)"
            />
          </g>
          <defs>
            <filter
              colorInterpolationFilters="sRGB"
              filterUnits="userSpaceOnUse"
              id="filter_:r0:"
            >
              <feFlood
                floodOpacity={0}
                result="BackgroundImageFix"
              />
              <feBlend
                in="SourceGraphic"
                in2="BackgroundImageFix"
                result="shape"
              />
              <feGaussianBlur
                result="effect1_foregroundBlur"
                stdDeviation={7}
              />
            </filter>
          </defs>
        </svg>
      </div>
      <p
        className="text-base font-medium undefined"
      >
        0x7F30
        ...
        5a347
      </p>
      <div
        className="w-4 h-4"
        onClick={[Function]}
      >
        <img
          alt="copy-adress-icon"
          className="cursor-pointer"
          src={
            {
              "ReactComponent": {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "default": "copy-icon.svg",
            }
          }
          style={
            {
              "height": "full",
              "width": "full",
            }
          }
        />
      </div>
    </div>
    <div
      className="mobile:border mobile:border-medium_grey mobile:rounded-[10px] mobile:p-4 mobile:w-full"
    >
      <p
        className="text-base font-medium text-purple_light mb-2"
      >
        title.totalBalance
      </p>
      <div
        className="flex gap-4 items-end"
      >
        <h1
          className="desktop:text-3xl desktop:leading-[45px] tablet:text-3xl tablet:leading-[45px] mobile:text-xl mobile:leading-5 font-medium text-[50px]"
        >
          $
          1000.00
        </h1>
        <div
          className="flex gap-1 items-center"
        >
          <img
            alt="percentage-up"
            className="w-[10px] h[9px]"
            src={
              {
                "ReactComponent": {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "default": "triangle-up-green.svg",
              }
            }
          />
          <p
            className="text-sm font-medium text-percentage_green"
          >
            1.00
            %
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex w-full flex-col items-end justify-end mobile:justify-normal gap-5"
  >
    <div
      className="flex w-full gap-4 justify-end mobile:flex-row tablet:flex-col tablet:items-end mobile:flex-row mobile:justify-between"
    />
  </div>
</div>
`;
